File: joiner-core/src/main/java/cz/encircled/joiner/util/Assert.java
package cz.encircled.joiner.util;

/**
 * @author Vlad on 14-Aug-16.
 */
public class Assert {

    public static void notNull(Object object) {
        if (object == null) {
            throw new IllegalArgumentException("Argument must be not null!");
        }
    }

    public static void assertThat(boolean predicate) {
        if (!predicate) {
            throw new IllegalArgumentException("Predicate must be true!");
        }
    }

}

File: joiner-core/src/main/java/cz/encircled/joiner/util/Assert.kt
package cz.encircled.joiner.util

/**
 * Assert util methods

 * @author Vlad on 14-Aug-16.
 */
object Assert {

    fun notNull(`object`: Any?) {
        if (`object` == null) {
            throw IllegalArgumentException("Argument must be not null!")
        }
    }

    fun assertThat(predicate: Boolean) {
        if (!predicate) {
            throw IllegalArgumentException("Predicate must be true!")
        }
    }

}
